### ISSUE TITLE
Hash can look like scientific notation

### ISSUE THREAD

**Issue by samedii** [guildai/guildai#50]
We had a small problem where a hash was interpreted by guild as scientific notation for a float.

Wanted to run `guild run evaluate model=3e563` but this was not interpreted correctly. Tried to specify that this was a string in the guild-file and in argparse but it did not work.

Note that commands like `guild export 3e563` do work correctly

**Comment by gar1t** [comment]
I've confirmed this is indeed a problem! Thank you for the report!

Here's a write up of the issue and a script to recreate it:

https://github.com/guildai/issues/tree/master/50-hash-can-look-like-scientific-notation

I'll get a fix for this into 0.6.6 - and will update here as soon as there's a pre-release you can use.

**Comment by samedii** [comment]
Cool, thank you!

**Comment by samedii** [comment]
This is an issue in dev8 still when running `guild run track_object model=67217e15`
It works if I provide the full hash instead.

Edit: Also does not work on dev9

```
ERROR: resolving required source 'operation:train|retrain' in model resource
Traceback (most recent call last):
  File "/home/richard/Documents/tracking/venv/lib/python3.6/site-packages/guild/deps.py", line 70, in resolve_source
    source_paths = resolver.resolve(unpack_dir)
  File "/home/richard/Documents/tracking/venv/lib/python3.6/site-packages/guild/resolver.py", line 236, in resolve
    source_path = self._source_path()
  File "/home/richard/Documents/tracking/venv/lib/python3.6/site-packages/guild/resolver.py", line 242, in _source_path
    if run_spec and os.path.isdir(run_spec):
  File "/home/richard/Documents/tracking/venv/lib/python3.6/genericpath.py", line 42, in isdir
    st = os.stat(s)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not float
guild: run failed because a dependency was not met: unexpected error resolving 'operation:train|retrain' in model resource: TypeError('stat: path should be string, bytes, os.PathLike or integer, not float',)

```

**Comment by gar1t** [comment]
This is fixed and pending release in 0.7.0.

The steps for demonstrating the correct behavior are here:

https://github.com/guildai/issue-resolution/tree/master/50-hash-can-look-like-scientific-notation

**Comment by gar1t** [comment]
The fix is now generally available in 0.7.0. If this is still an issue, please re-open and we'll sort it out. Thanks again for the report!
